internal/app/user-api/validation/user_validator.go

```diff
@@ -0,0 +1,78 @@
+package validation
+
+import (
+	"strings"
+
+	"github.com/afernandowex/ps-tag-onboarding-go/internal/app/model"
+	"github.com/afernandowex/ps-tag-onboarding-go/internal/app/user-api/constant"
+	"github.com/google/uuid"
+)
+
+type IUserValidationService interface {
+	ValidateUser(user *model.User) []string
+	ValidateUserID(ID *string) []string
+}
+
+type UserValidationService struct {
+}
+
+func (s *UserValidationService) ValidateUserID(ID *string) []string {
+	var errors []string
+	if ID == nil || !isUUID(*ID) {
+		errors = append(errors, constant.ErrorInvalidUserID)
+	}
+
+	return errors
+}
+
+func isUUID(str string) bool {
```

Comment:
Why did you write your own validation for uuid? It is basically just a nil check and parsing it.